Lib/pydoc.py

```diff
@@ -873,7 +873,7 @@ def spilldata(msg, attrs, predicate):
                 thisclass = attrs[0][2]
             attrs, inherited = _split_list(attrs, lambda t: t[2] is thisclass)
 
-            if thisclass is builtins.object:
+            if object is not builtins.object and thisclass is builtins.object:
```

Comment:
What is the purpose of this change?

Reply:
The have `help(object)` print the methods defined on `object`, the same as it does for every other class.

Reply:
One confusing thing about this module is that the argument that the docs are being created for is named is *object*.  So, for this PR, object both can refer to the name of the argument and the class called object.  

`thisclass` is the third value (index 2) in the sequence returned from `inspect.classify_class_attrs(object)`, which is defined as "The class which defined this attribute" (from the docstring).  When the value of the class is `object`, the current if statement `if thisclass is builtins.object:` results in that attribute being skipped from the output.

The change is to skip the logic when the argument value is not builtins.object so that the attributes will be displayed when `help(object)` is used.  Without this change, all of the attributes are skipped because "The class which defined this attribute" is always `object`.